crates/apollo-compiler/src/diagnostics.rs

```diff
@@ -265,6 +267,13 @@ pub enum DiagnosticData {
         /// Name of the field
         field: String,
     },
+    #[error("operation must not select different types as the same name `{field}`")]
+    ConflictingSelection {
```

Comment:
not entirely 100% on suggestion, but maybe!

```suggestion
    ConflictingFields {
```